pkg/controller/atlasdatafederation/datafederation_controller.go

```diff
@@ -260,3 +263,31 @@ func (r *AtlasDataFederationReconciler) Delete(e event.DeleteEvent) error {
 
 	return nil
 }
+
+func managedByAtlas(ctx context.Context, atlasClient mongodbatlas.Client, projectID string, log *zap.SugaredLogger) customresource.AtlasChecker {
+	return func(resource mdbv1.AtlasCustomResource) (bool, error) {
+		dataFederation, ok := resource.(*mdbv1.AtlasDataFederation)
+		if !ok {
+			return false, errors.New("failed to match resource type as AtlasDataFederation")
+		}
+
+		if dataFederation.Spec.Name == "" {
```

Comment:
This is unlikely to happen once it's a DataFederation can't exist without a name, but if you want to keep this check, better to return an error